### DISCUSSION TITLE
File Upload Display Name separate from Stored File Name

### DISCUSSION THREAD

**Discussion by GabrielSieben** [filamentphp/filament#4010] [Help]
Hello,

I'm building a fairly large application which will have many file uploads. For the sake of the user, saving the original name of the file uploaded (i.e. "Hello.png") is very important. However, storing it that way is not ideal for obvious reasons, such as potential conflicts.

The way I'm looking at is to use the storeFileNamesIn() to save the original file name, while keeping the file randomized on-disk in, say, a giant S3 folder.  However... there's no way, that I can find, to customize the displayed file name in the File Upload field. So, I might have the original file name on hand, but I can't actually tell Filament to display that stored file name instead of the randomly-generated one. 

Thoughts?

**Comment by ManojKiranA** [comment]
This feature is already available

![image](https://user-images.githubusercontent.com/30294553/190062911-3af7efbf-84e5-4767-a9fa-219bb680cea0.png)

**Reply by GabrielSieben** [reply]
I wish. I did know that existed, but that only stores the original file name in a column. However, if I use a filament form for editing a model, it shows the random name in the "File Upload" box instead of the original file name. I can't feed that original file name back into the form for display.

**Reply by danharrin** [reply]
We use Filepond to power the file upload field. As far as I know, Filepond will only display the label of the file as the actual file URL that we give it. So I don't think this is possible.

You could try the `preserveFilenames()` option, combined with our Spatie Media Library plugin. `preserveFilenames()` stores the files with their original names. Spatie Media Library stores the files in different directories based on the Eloquent record they are associated with. Since the file names are preserved, they should show correctly on the file upload field.

**Reply by GabrielSieben** [reply]
I will look into that further. When I was looking at I thought, it seemed that media library did best with an assortment of items, but was quite unwieldy if you only needed one image field? But if that’s what’s needed…

**Reply by danharrin** [reply]
It should work fine with just 1 image
